metrics_sdk/lib/opentelemetry/sdk/metrics/instrument/synchronous_instrument.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Instrument
+        class SynchronousInstrument
+          def initialize(name, unit, description, instrumentation_library, meter_provider)
+            @name = name
+            @unit = unit
+            @description = description
+            @instrumentation_library = instrumentation_library
+            @metric_streams = []
+
+            meter_provider.metric_readers.each do |metric_reader|
+              register_with_new_metric_store(metric_reader.metric_store)
```

Comment:
OIC - so you *do* need `metric_store` exposed on `MetricReader` 🤔 - maybe make it API-private?